### PR TITLE
macvlan: ARP proxy support

### LINKED ISSUE #84: macvlan: neighbours are not aware of ARP changes
I would expect for the kernel to take care of this, but somehow it doesn't seem to work if IPs are reused. This issue stems from https://github.com/coreos/rkt/issues/1765.

### PR DESCRIPTION
**Pull request by jellonek**
Fixes #84.

--
This PR provides small utility package to operate on sysctl on linux platform (counterpart to similar function in stdlib, but only accessible for *BSD based platforms) used in second patch - which provides proxy_arp setting on newly created macvlan interfaces following description in https://github.com/coreos/rkt/issues/1765

@steveeJ: review request.

### CODE REVIEW

**Comment by jellonek** [comment]
Looking on https://github.com/appc/cni/commit/f49145d5ff6b2c48fe243b74be240492ba75dec3 - at least path to file could be rewritten in similar way (use of filepath.Join and filepath.Clean) - i'll do that tomorrow.

File: pkg/util/sysctl/sysctl_linux.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+package sysctl
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+const (
```
Comment (steveeJ) [inline pkg/util/sysctl/sysctl_linux.go]: are these ever used?

**Comment by jellonek** [comment]
Review request.

**Comment by jellonek** [comment]
Note: i was asked on rkt community sync to add test which covers changes added in this PR.

**Comment by zachgersh** [comment]
Nice @jellonek - we'll keep an eye out for those tests.

**Comment by f0** [comment]
any chance to get this into the next rkt release?

**Comment by jellonek** [comment]
I cant add tests for whole plugin. If someone will add them for current version, I can add additional test covering changes from this PR.

File: pkg/util/sysctl/sysctl_linux.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+func Sysctl(name string, params ...string) (string, error) {
+	if len(params) > 1 {
+		return "", fmt.Errorf("unexcepted additional parameters")
+	} else if len(params) == 1 {
+		return set_sysctl(name, params[0])
+	}
+	return get_sysctl(name)
+}
+
+func get_sysctl(name string) (string, error) {
```
Comment (steveeJ) [inline pkg/util/sysctl/sysctl_linux.go]: please use camelCase here

File: pkg/util/sysctl/sysctl_linux.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	full_name := filepath.Join("/proc/sys", strings.Replace(name, ".", "/", -1))
```
Comment (steveeJ) [inline pkg/util/sysctl/sysctl_linux.go]: please use camelCase here

File: pkg/util/sysctl/sysctl_linux.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	full_name = filepath.Clean(full_name)
+	data, err := ioutil.ReadFile(full_name)
+	if err != nil {
+		return "", err
+	}
+
+	return string(data[:len(data)-1]), nil
+}
+
+func set_sysctl(name, value string) (string, error) {
```
Comment (steveeJ) [inline pkg/util/sysctl/sysctl_linux.go]: please use camelCase here

File: pkg/util/sysctl/sysctl_linux.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	full_name := filepath.Join("/proc/sys", strings.Replace(name, ".", "/", -1))
```
Comment (steveeJ) [inline pkg/util/sysctl/sysctl_linux.go]: please use camelCase here

**Comment by steveeJ** [comment]
Superseded by #149

**Pull request closed without merging**